src/plugins/core/data_validation.ts

```diff
@@ -192,16 +212,19 @@ export class DataValidationPlugin
     this.history.update("rules", sheetId, newRules);
   }
 
-  private setCenterStyleToBooleanCells(rule: DataValidationRule) {
+  private setStyleAndContentToBooleanCells(rule: DataValidationRule) {
     for (const position of getCellPositionsInRanges(rule.ranges)) {
       const cell = this.getters.getCell(position);
-      const { sheetId, col, row } = position;
       const style: Style = {
         ...cell?.style,
         align: cell?.style?.align ?? "center",
         verticalAlign: cell?.style?.verticalAlign ?? "middle",
       };
-      this.dispatch("UPDATE_CELL", { sheetId, col, row, style });
+      let updatedCell: Omit<UpdateCellCommand, "type"> = { ...position, style };
+      if (cell?.content !== "true" && cell?.content !== "TRUE") {
```

Comment:
```suggestion
      if (cell?.content.toLowerCase() !== "true") {
```
more robust if it's ever possible to have cell values be somthing like `TrUe`